src/builders/rust/utils.nix

```diff
@@ -15,17 +18,83 @@ in rec {
 
   # Generates a script that replaces relative path dependency paths with absolute
   # ones, if the path dependency isn't in the source dream2nix provides
-  replaceRelativePathsWithAbsolute = paths: let
+  replaceRelativePathsWithAbsolute = {paths}: let
     replacements =
       l.concatStringsSep
       " \\\n"
       (
         l.mapAttrsToList
-        (rel: abs: "--replace '\"${rel}\"' '\"${abs}\"'")
+        (
+          from: rel: ''--replace "\"${from}\"" "\"$TEMPDIR/$sourceRoot/${rel}\""''
+        )
         paths
       );
   in ''
     substituteInPlace ./Cargo.toml \
       ${replacements}
   '';
+
+  # Script to write the Cargo.lock if it doesn't already exist.
+  writeCargoLock = ''
+    if [ ! -f "$PWD/Cargo.lock" ]; then
+      cp -v ${cargoLock} "$PWD/Cargo.lock"
```

Comment:
This triggers the build of `cargoLock` to produce the file in the nix store. This will be counted as import from derivation, because nix has to build this derivation (via builtins.toFile) in order to continue the evaluation.

Alternatively you could insert the lock file as a string here, and therefore not forcing nix to create a store path during eval time.